### PR TITLE
Add libthrift

### PR DESCRIPTION
**Pull request by isaachier**
Thrift C++ library without compiler and dependencies.

### CODE REVIEW

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -219,6 +219,7 @@ hunter_config(libjson-rpc-cpp VERSION 0.7.0-p3)
 hunter_config(libmill VERSION 1.18)
 hunter_config(libogg VERSION 1.3.2-cmake3)
 hunter_config(libsodium VERSION 1.0.10)
+hunter_config(libthrift VERSION 0.10.0)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Two equal projects with the different names?

Comment (isaachier) [reply cmake/configs/default.cmake]: They pass in different cmake args. Also, because libthrift has no dependency on bison, it is cachable.

Comment (ruslo) [reply cmake/configs/default.cmake]: Both use URL `https://github.com/hunter-packages/thrift/archive/v0.10.0-p2.tar.gz` - hence they are the same packages.

> They pass in different cmake args

Can be controlled by custom `CMAKE_ARGS`:
* https://github.com/ruslo/hunter/wiki/example.custom.config.id#custom-cmake-options

`CMAKE_ARGS` in `hunter.cmake` just set some reasonable default values. The rest is up to user. Also see this feature request: https://github.com/ruslo/hunter/issues/1106

Comment (isaachier) [reply cmake/configs/default.cmake]: What about `hunter_cacheable` and the fact that libthrift supports many more platforms?

Comment (ruslo) [reply cmake/configs/default.cmake]: Set best `CMAKE_ARGS` configuration as a default and add warnings to documentation about other options. If cacheable/uncacheable depends on CMAKE_ARGS then user can break it just by specifying custom options. Having two packages that do install same files is a badly designed solution.

**Comment by isaachier** [comment]
OK I think I understand. One last question: why does Protobuf have 3 packages?

**Comment by ruslo** [comment]
> why does Protobuf have 3 packages?

What do you mean? I see only one:
* https://docs.hunter.sh/en/latest/packages/all.html?highlight=protobuf

**Comment by isaachier** [comment]
I guess you are right. Just three examples.

**Pull request closed without merging**